code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm

```diff
@@ -450,6 +448,15 @@
 	for(var/obj/item/potential_tinder in location)
 		potential_tinder.extinguish()
 
+/datum/effect_system/fluid_spread/foam/metal/resin/halon
+	effect_type = /obj/effect/particle_effect/fluid/foam/metal/resin/halon
+
+/// A variant of resin foam that is created from halon combustion. It does not dissolve in heat to allow the gas to spread before foaming.
+/obj/effect/particle_effect/fluid/foam/metal/resin/halon
+
+/obj/effect/particle_effect/fluid/foam/metal/resin/halon/atmos_expose(datum/gas_mixture/air, exposed_temperature)
+	return // Doesn't dissolve in heat.
```

Comment:
should override should_atmos_process and also maybe remove the element, otherwise this will process forever for no reason

Reply:
Sure.